lib/core/hooks.js

```diff
@@ -83,6 +83,19 @@ export async function buildHook (moduleContainer, options, logger) {
       }
     }
   }
+  if (options.tracing) {
+    options.tracing = deepMerge.all([{
+      tracesSampleRate: 1.0,
+      vueOptions: {
+        tracing: true,
+        tracingOptions: {
+          trackComponents: true
+        }
```

Comment:
Let's add all those (plus types):

```ts
/** Vue specific configuration for Tracing Integration  */
interface TracingOptions {
    /**
     * Decides whether to track components by hooking into its lifecycle methods.
     * Can be either set to `boolean` to enable/disable tracking for all of them.
     * Or to an array of specific component names (case-sensitive).
     */
    trackComponents: boolean | string[];
    /** How long to wait until the tracked root activity is marked as finished and sent of to Sentry */
    timeout: number;
    /**
     * List of hooks to keep track of during component lifecycle.
     * Available hooks: 'activate' | 'create' | 'destroy' | 'mount' | 'update'
     * Based on https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
     */
    hooks: Operation[];
}
declare type Operation = 'activate' | 'create' | 'destroy' | 'mount' | 'update';
```
Unfortunately, we can't import types directly from sentry so have to duplicate

Reply:
What should be used for the defaults for `timeout` and `hooks`? Looking at Sentry's source, timeout is `2000` and hooks are `mount` and `update` ?

https://github.com/getsentry/sentry-javascript/blob/master/packages/integrations/src/vue.ts#L164-L169

Reply:
Yeah. Let's use the same.

As much as I hate duplicating stuff and risking things getting out of sync, it makes it quite more transparent for the user in case he wants to modify some of those.